mendix/widgets-resources
# [N] Line chart - Add consistency checking

**Story**
WT-2584

**Why**
Line chart has conditional visible configuration properties. Therefore, some properties had to be set to `required=false` while in a specific mode they are actually required.

**What has been done**
- Add custom consistency checks to make sure all required properties (in a certain static and dynamic mode) are configured.
Bonus:
- Lock dependencies for line chart external lib victory (+ some other dependencies).
- Code improvements.
- Add page editor util function to hide multiple nested properties in piw-utils.
- Rename `X value` and `Y value` configuration property to `X attribute` and `Y attribute`.

**Important note**
- The configured check for data source configuration properties is implemented in such a way to deal with a bug in page editor. As soon as this is fixed, this check will have to be rewritten.

**How to test**
- Open the configuration window of the widget and check if all configuration are properly hidden for static and dynamic series. Also check if widget throws consistency error if certain configuration properties aren't configured.
